src/bufr/DataObject.h

```diff
@@ -657,6 +667,24 @@ namespace Ingester
             return 0.0f;
         }
 
+        /// \brief Get the data at the index as a int for numeric data.
+        /// \return Int data.
+        template<typename U = void>
+        std::string _getAsString(size_t idx,
+                                 typename std::enable_if<std::is_arithmetic<T>::value, U>::type* = nullptr) const
+        {
+            return std::to_string(data_[idx]);
+        }
+
+        /// \brief Get the data at the index as a int for non-numeric data.
+        /// \return Int data.
+        template<typename U = void>
+        std::string _getAsString(size_t idx,
+                                 typename std::enable_if<!std::is_arithmetic<T>::value, U>::type* = nullptr) const
+        {
+            return data_[idx];
```

Comment:
Just want to double check if this is safe. If I understand the code correctly, it is assuming you have a string type (or something that can automatically convert to string) if the is_arithmetic test returns false which might be problematic. However, I think it's okay if you only expect integer, float and string types in the data_ variable.

Reply:
So I thought this would be OK since the compiler should refuse to compile the DataObject if it can't convert the data_ (of type T) to a string as the return value is explicitly required to be a std::string.

Reply:
I was going to refactor the DataObject as part of porting the code to IODA as well... I was going to use template specialization instead of just straight SFINAE in order toe draw a clearer distinction between string and numeric types...

Reply:
@rmclaren thanks for explaining this.

Reply:
@rmclaren thanks for the explanation! I think it all sounds good.